### PR TITLE
[integrations][CrowdStrike] - Fixed event tag handling for the falcon data-stream

### PR DESCRIPTION
**Pull request by ShourieG**
## Type of change
- Bug

## What does this PR do?
It was noted by many users that often falcon data-stream logs get dropped due to a tag parsing issue. The field in question here is the **crowdstrike.event.Tags**. It was observed from client feedback from SDHs and also personal testing that the value of this field can sometimes be an array of objects, a comma separated string or a normal string. 

**What existed in the pipeline:**  
The pipeline defined the field **Tags** as of type **nested**. This obviously caused issues when the value would be a string or a comma separated string as they would result in mapping errors when pushed to elasticsearch.

**What was the solution:** 
1) We decided to consolidate all the **Tags** under **ctx.tags** which resides at the document root. This would avoid 
     having different tag fields for different type of **Tags**.
2) If **crowdstrike.event.Tags** is an array of objects we concatenate the "Key" and "ValueString" which are the only two 
     attributes per object into a string format "Key:ValueString" and append it to **ctx.tags**.
3)  If **crowdstrike.event.Tags** is a comma separated string, we split it and transform it into a List and append each 
      element of the list to **ctx.tags**
4) If **crowdstrike.event.Tags** is a standard string, we simply append it  to **ctx.tags**.
5) Finally we remove the field **crowdstrike.event.Tags**

This ensures consistent tag handling across the data-stream for all type of logs.

## Checklist
- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ]

## Related issues

-

### CODE REVIEW

File: packages/crowdstrike/data_stream/falcon/elasticsearch/ingest_pipeline/default.yml [hunk 1]
```diff
@@ -78,6 +78,25 @@ processors:
                     }
                 } 
             }
+  # Handle event Tags, which can be a string or a list of maps.
+  - script:
+      tag: convert-Tags-and-copy-to-tags
+      description: Convert tags from nested type and append to ctx.tags.
+      lang: painless
+      if : ctx.crowdstrike?.event != null && ctx.crowdstrike.event.Tags != null
```
Comment (kcreddy) [inline packages/crowdstrike/data_stream/falcon/elasticsearch/ingest_pipeline/default.yml:86]:
```suggestion
      if : ctx.crowdstrike?.event?.Tags != null
```

I think we can consolidate into 1 expression.

File: packages/crowdstrike/data_stream/falcon/_dev/test/pipeline/test-falcon-tags-list.log-expected.json [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+            "message": "EC2 NACL configured for global ingress",
+            "observer": {
+                "product": "Falcon",
+                "vendor": "Crowdstrike"
+            },
+            "tags": [
+                "preserve_original_event",
+                "SensorGroupingTags/TEACHER",
+                " SensorGroupingTags/XYZ",
+                " 321",
```
Comment (kcreddy) [inline packages/crowdstrike/data_stream/falcon/_dev/test/pipeline/test-falcon-tags-list.log-expected.json:67]: You could consider removing white spaces

**Review by kcreddy — APPROVED** [review]
Minor suggestions. Rest LGTM 👍🏼

File: packages/crowdstrike/data_stream/falcon/_dev/test/pipeline/test-falcon-tags-list.log-expected.json [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+                "service": {
+                    "name": "EC2"
+                }
+            },
+            "crowdstrike": {
+                "event": {
+                    "Finding": "Instance ID: i-0108fce80e5ab5129|VPC ID: vpc-0e886040c27d9f526|Network ACL ID: acl-005e6bb98e75ac17e|Rule Number: 100|CIDR Block: 0.0.0.0/0|Protocol: All",
+                    "PolicyId": 26,
+                    "ResourceAttributes": {
+                        " ACL ID": "acl - 005e6 bb98e75ac17e",
```
Comment (bhapas) [inline packages/crowdstrike/data_stream/falcon/_dev/test/pipeline/test-falcon-tags-list.log-expected.json:20]:
```suggestion
                        "ACL ID": "acl - 005e6 bb98e75ac17e",
```

I think it is just in the [test input](https://github.com/elastic/integrations/pull/7625/files#diff-d59dcbf2fafc17dd0bdf402eca886ed6527d6829a4901987d730b1b3ebda17d2R26)

**Review by bhapas — APPROVED** [review]
Minor comment. LGTM otherwise

**Pull request merged**